src/main/java/org/folio/circulation/resources/RequestByInstanceIdResource.java

```diff
@@ -364,33 +358,63 @@ private void createInstanceLevelRequests(RoutingContext routingContext) {
   }
 
   private Result<InstanceRequestRelatedRecords> combineWithUnavailableItems(
-    Result<Map<Item, DateTime>> itemDueDateMapResult,
-    Result<Map<Item, Integer>> itemRequestQueueSizeMapResult,
+    Map<Item, DateTime> itemDueDateMap,
     InstanceRequestRelatedRecords records){
 
     return of(() -> {
-      if (itemDueDateMapResult.succeeded() && itemRequestQueueSizeMapResult.succeeded()) {
-        Map<Item, DateTime> itemDueDateMap = itemDueDateMapResult.value();
-        Map<Item, Integer> itemQueueSizeMap = itemRequestQueueSizeMapResult.value();
+        Map<Item, RequestQueue> itemQueueMap = records.getItemRequestQueueMap();
 
-        if (itemDueDateMap == null && itemQueueSizeMap == null) {
+        if (itemDueDateMap == null && itemQueueMap == null) {
           return records;
         }
 
-        //Sort the map
+        //transform itemQueueMap to itemQueueSizeMap
+        Map<Item, Integer> itemQueueSizeMap = itemQueueMap.entrySet()
+          .stream()
+          .collect(Collectors.toMap(
+            Map.Entry::getKey,
+            entry -> entry.getValue().size()
+            ));
+
+        //Sort the itemQueueSize map
         Map<Item, Integer> sortedMap = sortRequestQueues(itemQueueSizeMap, itemDueDateMap,
           records.getInstanceLevelRequest().getPickupServicePointId());
 
         ArrayList<Item> finalOrdedList = new ArrayList<>(sortedMap.keySet());
 
         //put the items that it wasn't able to retrieve loans or RequestQueues for on the bottom of the list
-        finalOrdedList.addAll(records.getItemsWithoutLoans());
-        finalOrdedList.addAll(records.getItemsWithoutRequests());
+        if (records.getItemsWithoutLoans() != null)
+          finalOrdedList.addAll(records.getItemsWithoutLoans());
+        if (records.getItemsWithoutRequests() != null)
+          finalOrdedList.addAll(records.getItemsWithoutRequests());
         records.setSortedUnavailableItems(finalOrdedList);
-      }
       return records;
     });
   }
+
+  private Result<Collection<Item>> validateRequester(Map<Item, RequestQueue> itemRequestQueueMap, UUID requesterId) {
+    if (!itemRequestQueueMap.isEmpty()) {
+      Collection<RequestQueue> requestQueues = itemRequestQueueMap.values();
+      for (RequestQueue queue : requestQueues) {
+        final Optional<Request> matchingRequest = queue.getRequests()
+          .stream()
+          .filter(request -> request.isOpen()
+                           && Objects.equals(request.asJson().getString(REQUESTER_ID_FIELD),
+                                             requesterId.toString())
+          ).findFirst();
+
+        if (matchingRequest.isPresent()) {
+            Map<String, String> parameters = new HashMap<>();
+            parameters.put(REQUESTER_ID_FIELD, requesterId.toString());
+            parameters.put(ITEM_ID_FIELD, matchingRequest.get().asJson().getString(ITEM_ID_FIELD));
```

Comment:
If we still want to use the itemId here, can we change this to `matchingRequest.get().getItemId()` instead? This will save us from doing the `copy()` in `asJson()`.